smoothie.js

```diff
@@ -921,48 +899,61 @@
       // Draw the line...
       context.beginPath();
       // Retain lastX, lastY for calculating the control points of bezier curves.
-      var firstX = 0, firstY = 0, lastX = 0, lastY = 0;
+      var firstX = 0, lastX = 0, lastY = 0;
       for (var i = 0; i < dataSet.length && dataSet.length !== 1; i++) {
         var x = timeToXPixel(dataSet[i][0]),
             y = valueToYPixel(dataSet[i][1]);
 
         if (i === 0) {
           firstX = x;
-          firstY = y;
-          context.moveTo(x, y);
+          if (typeof dataSet[i][1] === 'string' ) {
+            context.moveTo(x, 0);
+            context.lineTo(x, dimensions.height);
+            context.fillStyle = seriesOptions.strokeStyle;
+            context.fillText(dataSet[i][1], x + 5, 10);
+          } else {
+            context.moveTo(x, y);
+          }
```

Comment:
Rather than do this type check on every point on every series, it'd be clearer and faster to create a new series type for these events.

Reply:
Wouldn't the function be exactly the same ? I just think it doubles the code unnecessarily

Reply:
The final outcome would be the same, but it's a lot more work to test the type on every single point on all series. Smoothie is light and fast. Currently this cost would be paid by _everyone_, regardless of whether they use these events. Using a new `SmoothieEventSeries` type avoids that cost.